### PR TITLE
Make some feature-generated links load via AJAX

### PR DESCRIPTION
**Pull request by kidonng**
Motivation: https://github.com/sindresorhus/refined-github/pull/4631#issuecomment-891201750

This PR will continue for a few days in case anyone find something new, but hopefully searching for `<a` will give not too much results to check.

Ajaxed features and Test URLS:

- `comments-time-machine-links`: https://github.com/sindresorhus/refined-github/blob/main/.gitattributes?rgh-link-date=2021-08-01
- `linkify-branch-references`: 🤷‍♀️ Something like https://github.com/sindresorhus/refined-github/compare/main...test?quick_pull=1
- `linkify-symbolic-links`: https://github.com/lxqt/lxqt/blob/master/CONTRIBUTING
- `list-prs-for-file`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix
- `rgh-feature-descriptions`: https://github.com/sindresorhus/refined-github/blob/main/source/features/ci-link.tsx
- `swap-branches-on-compare`: https://github.com/sindresorhus/refined-github/compare/hotfix...main
- `view-markdown-source`: https://github.com/sindresorhus/refined-github/blob/main/readme.md

### CODE REVIEW

File: source/features/comments-time-machine-links.tsx [hunk 1]
```diff
@@ -105,7 +105,7 @@ async function showTimemachineBar(): Promise<void | false> {
 	}
 
 	addNotice(
-		<>You can also <a className="rgh-link-date" href={String(url)}>view this object as it appeared at the time of the comment</a> (<relative-time datetime={date}/>)</>,
+		<>You can also <a className="rgh-link-date" href={String(url)} data-pjax="#repo-content-pjax-container">view this object as it appeared at the time of the comment</a> (<relative-time datetime={date}/>)</>,
```
Comment (yakov116) [inline source/features/comments-time-machine-links.tsx:108]:
```suggestion
		<>You can also 
			<a 
				className="rgh-link-date"
				href={String(url)}
				data-pjax="#repo-content-pjax-container"
			>
			view this object as it appeared at the time of the comment
			</a> (<relative-time datetime={date}/>)</>,
```

Too many things to be on one line. (Don't commit this its going to fail linting)

Comment (kidonng) [reply source/features/comments-time-machine-links.tsx:108]: I was wondering why linter doesn't give warnings on these super long lines

Comment (yakov116) [reply source/features/comments-time-machine-links.tsx:108]: We don't have an issue with the long line so much. The issue is having so many attributes on the same line. It gets hard to read.

Comment (fregante) [reply source/features/comments-time-machine-links.tsx]: What I do sometimes is extract the JSX out so it's a single variable. e.g. 
```jsx
const link = (
	<a className="rgh-link-date" href={String(url)} data-pjax="#repo-content-pjax-container">
		view this object as it appeared at the time of the comment
	</a>
);
```
```suggestion
		<>You can also {link} (<relative-time datetime={date}/>)</>,
```

File: source/features/list-prs-for-file.tsx [hunk 2]
```diff
@@ -34,7 +34,7 @@ function getDropdown(prs: number[]): HTMLElement {
 						data-url={buildRepoURL('issues', prNumber)}
 						data-id={`rgh-pr-${prNumber}`}
 					>
-						<a className="dropdown-item" href={getPRUrl(prNumber)}>
+						<a className="dropdown-item" href={getPRUrl(prNumber)} data-pjax="#js-repo-pjax-container">
```
Comment (kidonng) [inline source/features/list-prs-for-file.tsx:37]: If `#repo-content-pjax-container` is used GitHub will 304 redirect to return `#js-repo-pjax-container"` which breaks the page, so just use the latter directly

**Comment by kidonng** [comment]
I have been focusing on some other stuff lately so I didn't really get to skim through every feature. I will try to do that tomorrow or the day after that.

**Comment by fregante** [comment]
@kidonng @yakov116 we can merge this now if the current ones have been tested and:

- they actually change the type of navigation
- they don’t introduce immediately-visible bugs (e.g. if you click and go back and go forward and everything is still fine then we’re good)

**Comment by fregante** [comment]
I've been seeing more and more often that features just don't work/appear on PRs and I have to refresh the page. Are you see thing issue? Does this fix the issue?

**Comment by kidonng** [comment]
There's too many features and probably I won't notice anything missing unless it's severely broken.

This PR should only deal with links added by Refined GitHub and should have nothing to do with the issue.

**Comment by fregante** [comment]
Has this been tested? Do you want to merge it as is?

**Comment by kidonng** [comment]
Sorry for creating another long-standing PR 😅  now that you mentioned I will _certainly_ push some more changes later today

**Comment by kidonng** [comment]
There are some links I haven't go through and I may add something new later but this PR is mergeable anytime.

**Comment by fregante** [comment]
Thanks! We won't release a new version for probably a week so we can see any issues while using the locally-built version.

**Pull request merged**